execution_engine/src/core/runtime_context/mod.rs

```diff
@@ -758,8 +764,16 @@ where
     /// Checks if we are calling a system contract.
     pub(crate) fn is_system_contract(&self) -> bool {
         if let Some(hash) = self.base_key().into_hash() {
-            let system_contracts = self.protocol_data().system_contracts();
-            if system_contracts.contains(&hash.into()) {
+            let system_contracts_registry = self
+                .tracking_copy
+                .borrow_mut()
+                .get_system_contracts(self.correlation_id)
+                .expect("should have registry");
+
+            if system_contracts_registry
+                .into_values()
+                .any(|system_hash| system_hash == hash.into())
+            {
                 return true;
             }
```

Comment:
If you rename `fn get_system_contract_registry()` as suggested in a subsequent comment, you can change this to:
```suggestion
            let contract_hash = ContractHash::new(hash);
            return self
                .system_contract_registry()
                .values()
                .any(|&system_hash| system_hash == contract_hash);
```

Reply:
Applied suggestion